c17032740.lua

```diff
@@ -51,49 +51,79 @@ end
 function c17032740.splimit(e,se,sp,st)
 	return not e:GetHandler():IsLocation(LOCATION_EXTRA)
 end
-function c17032740.spfilter(c,code)
-	return c:IsAbleToDeckOrExtraAsCost() and c:IsFusionCode(code)
+function c17032740.spfilter(c,code1,code2,code3)
+	return c:IsAbleToDeckOrExtraAsCost() and (c:IsFusionCode(code1) or c:IsFusionCode(code2) or c:IsFusionCode(code3))
+end
+function c17032740.spfilter1(c,mg)
+	local mg2=mg:Clone()
+	mg2:RemoveCard(c)
+	return c:IsFusionCode(89943723) and c:IsAbleToDeckOrExtraAsCost() and c:IsCanBeFusionMaterial()
+		and mg2:IsExists(c17032740.spfilter2,1,nil,mg2)
+end
+function c17032740.spfilter2(c,mg)
+	local mg2=mg:Clone()
+	mg2:RemoveCard(c)
+	return c:IsFusionCode(43237273) and c:IsAbleToDeckOrExtraAsCost() and c:IsCanBeFusionMaterial()
+		and mg2:IsExists(c17032740.spfilter3,1,nil)
+end
+function c17032740.spfilter3(c)
+	return c:IsFusionCode(17732278) and c:IsAbleToDeckOrExtraAsCost() and c:IsCanBeFusionMaterial()
+end
+function c17032740.rmfilter(c,code)
+	return c:IsFusionCode(code) and c:IsCode(code)
 end
 function c17032740.spcon(e,c)
 	if c==nil then return true end
 	local tp=c:GetControler()
 	local ft=Duel.GetLocationCount(tp,LOCATION_MZONE)
 	if ft<-2 then return false end
-	local g1=Duel.GetMatchingGroup(c17032740.spfilter,tp,LOCATION_ONFIELD,0,nil,89943723)
-	local g2=Duel.GetMatchingGroup(c17032740.spfilter,tp,LOCATION_ONFIELD,0,nil,43237273)
-	local g3=Duel.GetMatchingGroup(c17032740.spfilter,tp,LOCATION_ONFIELD,0,nil,17732278)
-	if g1:GetCount()==0 or g2:GetCount()==0 or g3:GetCount()==0 then return false end
-	if ft>0 then return true end
-	local f1=g1:FilterCount(Card.IsLocation,nil,LOCATION_MZONE)>0 and 1 or 0
-	local f2=g2:FilterCount(Card.IsLocation,nil,LOCATION_MZONE)>0 and 1 or 0
-	local f3=g3:FilterCount(Card.IsLocation,nil,LOCATION_MZONE)>0 and 1 or 0
-	if ft==-2 then return f1+f2+f3==3
-	elseif ft==-1 then return f1+f2+f3>=2
-	else return f1+f2+f3>=1 end
+	local mg=Duel.GetMatchingGroup(c17032740.spfilter,tp,LOCATION_ONFIELD,0,nil,89943723,43237273,17732278)
+	if ft>0 then return mg:IsExists(c17032740.spfilter1,1,nil,mg) end
+	local mg2=mg:Filter(Card.IsLocation,nil,LOCATION_MZONE)
+	local ct=mg2:GetClassCount(Card.GetFusionCode)
```

Comment:
Since `Card.GetFusionCode` has multiple return value, it's not suitable to use `GetClassCount`, which only compare the first return value.